NAS2D/Renderer/Renderer.cpp

```diff
@@ -102,12 +77,6 @@ void Renderer::fadeIn(std::chrono::milliseconds fadeInTime)
 }
 
 
-/**
- * Non-blocking screen fade.
- *
- * \param	fadeOutTime	Time in miliseconds the fade should last. A value of 0
- *					will instantly fade the screen in.
```

Comment:
Ditto.

Reply:
The fade functions are most-definitely blocking. They just set up variables to be updated in `Renderer::Update` which runs on the main thread:

```cpp
void Renderer::update()
{
	if (mCurrentFadeType != FadeType::None)
	{
		float fade = (fadeTimer.delta() * mFadeStep) * static_cast<int>(mCurrentFadeType);

		mCurrentFade += fade;

		if (mCurrentFade < 0.0f || mCurrentFade > 255.0f)
		{
			mCurrentFade = std::clamp(mCurrentFade, 0.0f, 255.0f);
			mCurrentFadeType = FadeType::None;
			fadeCompleteSignal();
		}
	}

	if (mCurrentFade > 0.0f)
	{
		drawBoxFilled(Rectangle<float>::Create({0, 0}, size().to<float>()), mFadeColor.alphaFade(static_cast<uint8_t>(mCurrentFade)));
	}
}
```

As you can see, no where is a thread used or synchronized, nor is the value of the fade time passed to any thread. Therefore, the "Non-blocking screen fade" comments are at best out-of-date and at worst misleading. Together with the use of properly typed and named variables the entire block can be removed.

Reply:
I think we have a disagreement over what "non-blocking" means.

Nowhere in that function is there any code that would prevent it from returning for an unspecified amount of time. It does it's thing, and then returns. There is no waiting.

Reply:
I was conflating "non-blocking" with "asynchronous". Yes, these are "non-blocking", but so is any function by default. Specifically commenting this one as non-blocking is superfluous.

Reply:
Well, not exactly. Many function calls are blocking. A file read method is typically blocking, as disk IO may take arbitrarily long, particularly with disk read errors and recovery.

A fade method very well could be blocking, if it chose to do the entire fade before returning, along with timing delays. In this case, it merely sets up the parameters to do the fade, which then takes place during the `Renderer::update` method, and those updates are only one frame per call, with no built in timing delay.

We might debate the value of the comment, though I wouldn't exactly dismiss it as entirely superfluous.